OpenRA.Mods.Common/Traits/DamagedByTerrain.cs

```diff
@@ -9,17 +9,24 @@
  */
 #endregion
 
+using System.Collections.Generic;
 using System.Linq;
 using OpenRA.GameRules;
 using OpenRA.Traits;
 
 namespace OpenRA.Mods.Common.Traits
 {
 	[Desc("This actor receives damage from the given weapon when on the specified terrain type.")]
-	class DamagedByTerrainInfo : UpgradableTraitInfo, IRulesetLoaded, Requires<HealthInfo>
+	class DamagedByTerrainInfo : UpgradableTraitInfo, Requires<HealthInfo>
 	{
-		[Desc("The weapon which is used to damage the actor.")]
-		[WeaponReference, FieldLoader.Require] public readonly string Weapon;
+		[Desc("Amount of damage received per DamageDelay ticks.")]
+		[FieldLoader.Require] public readonly int Damage = 0;
+
+		[Desc("Delay between receiving damage.")]
+		public readonly int DamageDelay = 0;
```

Comment:
This is an interval, not a delay.  Please rename to `DamageInterval` for consistency.